projects/uvis_gs_fail/data_process.py

```diff
@@ -0,0 +1,238 @@
+# Imports:
+import os
+import sys
+from multiprocessing import Pool
+import numpy as np
+import pandas as pd
+from astropy.io import fits
+import tqdm
+from skimage.transform import resize, rotate
+
+
+
+def log_image_process(fpath, orig_data=False):
+    '''
+    Opens a flt.fits file and logarithmically processes the data according to 
+    the following method:
+    - opens the flt file, extracts SCI array, and creates an array of image 
+    data.
+    - Sets all values in the image array that are less than one to be equal to 
+    1.
+    - Scales the image data logarithmically.
+    - Uses min/max scaling to scale the pixel values to be between 0 and 1. 
+    - Resizes the image array to the dimensions (256, 256).
+
+    Parameters:
+    -----------
+    fpath: string
+    The string that identifies the directory that the image is located in.
+
+    Returns: numpy array of floats
+    --------
+    log_data: array of floats
+    Array of logarithmically scaled image data.
+
+    rootname: string
+    A String containing the rootname of the image that was processed.
+    '''
+
+     #Unpacking the image data: 
+        # get the boolean value of the subarray
+    subarray_value = fits.getval(fpath, 'subarray')
+    
+    # If the subarray is false then it is full chip, so we get the 
+    # two data arrays, minus the 6 dead rows of pixels
+    if subarray_value == False:
+        data1 = fits.getdata(fpath,'sci', 1)[:-3]
+        data2 = fits.getdata(fpath, 'sci', 2)[3:]
+        
+        image_data = np.vstack((data1, data2))
+        
+    else:
+        
+        image_data = fits.getdata(fpath, 'sci', 1)
+    
+        # Clip low values 
+    image_copy = image_data.copy() # Copy image_data array
+
+    pos_mask = image_copy < 1 # make positive mask
+    image_copy[pos_mask] = 1 # Make all values positive
+        
+        # log scale data 
+    log_data = np.log10(image_copy)
+            
+        # Resize log scaled image 
+    log_data = resize(log_data, (256,256), order=3) 
+    
+        # Min/max scale image array:
+    scaled_image = (log_data - log_data.min())/(log_data.max() 
+                                                  - log_data.min())
+    
+    if orig_data == True:
+        return(scaled_image, image_data)
+    else:
+        return(scaled_image)
+    
+    
+def augment(image):
+    '''
+    Takes in an array of image data, and performs data augmentation. The
```

Comment:
Performs data augmentation on image data. (an "an array of image data" sounds 3d (# of images, height width) but I think the object is 2d (height, width))